cmd/example-tests/runtest.go

```diff
@@ -0,0 +1,139 @@
+package main
+
+import (
+	"fmt"
+	"io"
+	"os"
+	"regexp"
+	"strings"
+	"time"
+
+	"github.com/onsi/ginkgo/v2"
+	"github.com/onsi/ginkgo/v2/types"
+
+	_ "github.com/openshift-eng/openshift-tests-extension/test/example"
+)
+
+// TestOptions handles running a single test.
+type TestOptions struct {
+	Out    io.Writer
+	ErrOut io.Writer
+}
+
+var _ ginkgo.GinkgoTestingT = &TestOptions{}
+
+func NewTestOptions(out io.Writer, errOut io.Writer) *TestOptions {
+	return &TestOptions{
+		Out:    out,
+		ErrOut: errOut,
+	}
+}
+
+func (opt *TestOptions) Run(args []string) error {
+	if len(args) != 1 {
+		return fmt.Errorf("only a single test name may be passed")
+	}
+
+	// Ignore the upstream suite behavior within test execution
+	ginkgo.GetSuite().ClearBeforeAndAfterSuiteNodes()
+	tests := testsForSuite()
+	var test *TestCase
+	for _, t := range tests {
```

Comment:
Hoping we can get most of the boilerplate in our module implementation. Where the test contributor implements a minimal Run(args []strings), then calls a pkg/ function in the module implementation like ExtensionRunImpl(testWrappers []ExtensionTestDescription, args []string). And the module takes it from there. 
That allows us the maximum level of control for subsequent versions of the module (e.g. how to construct ExtensionTestDescription, parameters that are interpreted by Run).